azure/scope/cluster.go

```diff
@@ -395,7 +398,11 @@ func (s *ClusterScope) SubnetSpecs() []azure.ResourceSpecGetter {
 			SecurityGroupName: azureBastionSubnet.SecurityGroup.Name,
 			RouteTableName:    azureBastionSubnet.RouteTable.Name,
 			Role:              azureBastionSubnet.Role,
-		})
+		}
+		if s.IsVnetManaged() {
```

Comment:
Why the differentiation between BYO vnet and managed vnet here? The entire reconciliation should be skipped if the vnet is not managed, and this differs from what we're doing for the other fields

Reply:
If the entire reconciliation is skipped if it isn't managed then this conditional can be removed. I'll update